### PR TITLE
Improve instructions for building mavlink custom messages

### PR DESCRIPTION
**Pull request by hamishwillee**
Fixes #1741 

@bkueng This does basic fixes to mavlink doc for PX4, explaining how messages are build and how they are tested.
It is a bit abstract as I have not actually tried it yet. 

Would appreciate your sanity check on the flow and concepts.

### CODE REVIEW

File: en/middleware/mavlink.md [hunk 1]
```diff
@@ -14,22 +14,28 @@ The tutorial assumes you have a [custom uORB](../middleware/uorb.md) `ca_traject
...
 
+PX4 includes the [mavlink/mavlink](https://github.com/mavlink/mavlink) repo as a submodule under [/src/modules/mavlink](https://github.com/PX4/PX4-Autopilot/tree/master/src/modules/mavlink), and generates the MAVLink 2 C library at build time from [/mavlink/messages/1.0/development.xml](https://github.com/mavlink/mavlink/blob/master/message_definitions/v1.0/development.xml) (by default).
+The dialect that is built is specified using the variable `MAVLINK_DIALECT` in [/src/modules/mavlink/CMakeLists.txt](https://github.com/PX4/PX4-Autopilot/blob/master/src/modules/mavlink/CMakeLists.txt#L34).
+
+In order to add your message we recommend that you create your messages in a new dialect file in the same directory, for example `PX4-Autopilot/src/modules/mavlink/mavlink/message_definitions/v1.0/custom_messages.xml`, and set `MAVLINK_DIALECT` to build the new file.
+This dialect file should include `development.xml`.
+
+:::note
+You might alternatively add your messages to `common.xml` or `development.xml`.
+Either way, remember that the same dialect file must be eventually be built in QGroundControl (or whatever software you use to communicate with PX4).
```
Comment (bkueng) [inline en/middleware/mavlink.md:25]:
```suggestion
Either way, remember that the same dialect file must eventually be built in QGroundControl (or whatever software you use to communicate with PX4).
```

File: en/middleware/mavlink.md [hunk 2]
```diff
@@ -14,22 +14,28 @@ The tutorial assumes you have a [custom uORB](../middleware/uorb.md) `ca_traject
...
-- [Generating MAVLink Libraries](https://mavlink.io/en/getting_started/generate_libraries.html)
+- [Install MAVLink Toolchain](https://mavlink.io/en/getting_started/installation.html)
+- [Generating MAVLink Libraries](https://mavlink.io/en/getting_started/generate_libraries.html).
 
-Your message needs to be generated as a C-library for MAVLink 2.
-Once you've [installed MAVLink](https://mavlink.io/en/getting_started/installation.html) you can do this on the command line using the command:
+Even though PX4 will build the files for you, you should test that your messages are correctly defined by building your XML file on the command line:
 ```sh
-python -m pymavlink.tools.mavgen --lang=C --wire-protocol=2.0 --output=generated/include/mavlink/v2.0 message_definitions/v1.0/custom_messages.xml
+python -m pymavlink.tools.mavgen --lang=C --wire-protocol=2.0 --output=generated/include/mavlink/v2.0 message_definitions/v1.0/https://github.com/mavlink/mavlink/blob/master/message_definitions/v1.0/development.xml
```
Comment (bkueng) [inline en/middleware/mavlink.md:36]: Simpler would be to just check the generated files in the build directory. However I don't think it's typically needed, unless something isn't working and you need to debug it.

**Review by bkueng — APPROVED** [review]

File: en/middleware/mavlink.md (hunk 2 above)
Comment (hamishwillee) [reply en/middleware/mavlink.md:36]: My thinking was that building in mavlink would simplify debugging if there were problems - for example, understanding how to format messages and seeing the validation errors. I'll rework though.

**Comment by hamishwillee** [comment]
Thanks very much @bkueng. I'll merge and iterate.

A related question in #1739 - it looks like the tutorial redefines  `_flow_pub` - the suggestion is to rename this in the tutorial to `_ca_traj_msg_pub` - so `uORB::Publication<ca_trajectory_s> _flow_pub{ORB_ID(ca_trajectory)};` changes to `uORB::Publication<ca_trajectory_s> _ca_traj_msg_pub{ORB_ID(ca_trajectory)};`

That makes sense to me - both as a solution and a normal pattern for naming to avoid clashes. Seem OK?

**Comment by bkueng** [comment]
> Seem OK?

Certainly

**Pull request merged**